Form with decision tree

mmircea16 opened the pull request:
### What changed?
 - Copied form from CeMaFac (with the exception of multiple choice type as it's not used)
 - Extracted logic around what to show at the end outside the form
 - Introduced the concept of nextQuestionId on option to build a decision point
 - Disabling the forward button when the choice is not made
 - Added basic test

This is work in progress but any feedback is more than welcomed. 

Eventually it will close: https://github.com/code4romania/taskforce-fe-components/issues/106

surdu commented on src/components/form/form.js:
```diff
@@ -0,0 +1,182 @@
+    return {
+      formId: data.formId,
+      timestamp: Date.now(),
+      answers: _.values(answersById)
```
When possible, please use native solutions instead of libraries.

```suggestion
      answers: Object.values(answersById)
```

surdu commented on src/components/form/form.js:
```diff
@@ -0,0 +1,182 @@
+    const selectedOption = _.find(
+      options,
+      option => option.value === optionValue
+    );
```
```suggestion
    const selectedOption = options.find(
      option => option.value === optionValue
    );
```

surdu commented:
Really awesome work! +💯 for also writing tests 😁

moonflare commented on src/components/form/results.js:
```diff
@@ -0,0 +1,39 @@
+      </div>
+    </div>
+  );
+}
```
We have now a component for this. We can remove this one. Please use [this](https://taskforce-fe-compone-git-fork-mmircea16-form-with-decisi-2be921.code4ro.now.sh/?path=/story/listitem--with-non-option-state) one

moonflare replied:
Also, please check [this](https://github.com/code4romania/ce-ma-fac/pull/164/files) PR for inspiration. There's been some nice updates for this part of the component.

moonflare commented on src/components/form/form.scss:
```diff
@@ -0,0 +1,25 @@
+    cursor: initial !important;
+    background-color: whitesmoke !important;
+  }
+}
```
We can also remove this one when we start using [this](https://taskforce-fe-compone-git-fork-mmircea16-form-with-decisi-2be921.code4ro.now.sh/?path=/story/listitem--with-non-option-state) component

moonflare commented on src/components/form/form.js:
```diff
@@ -0,0 +1,182 @@
+    return selectedOption.nextQuestionId;
+  };
+  const goToNextQuestion = () => {
+    // TODO use the disabled prop once the Button component implements it
```
We have [disabled](https://taskforce-fe-compone-git-fork-mmircea16-form-with-decisi-2be921.code4ro.now.sh/?path=/story/button--disabled) state for the button now

moonflare commented on src/components/form/form.js:
```diff
@@ -0,0 +1,182 @@
+import "./form.scss";
+import { Button } from "../button/button";
+import Results from "./results";
+import _ from "underscore";
```
I would try to import only what we need, and not everything. This way we reduce the amount of bundled JavaScript we export. Something like this. Not really familiar with underscore, but I know other libraries like `lodash` have it working this way.

```js
import groupBy from "underscore/groupBy";
import mapObject from "underscore/mapObject";
...etc
```

mmircea16 commented:
Thanks for your reviews!! I have implemented your suggestions (all good points!) Let me know if I missed something or there is something else to improve

Otherwise I have added the a "Free Text" type. That should complete the requirements for  https://github.com/code4romania/taskforce-fe-components/issues/106, I guess
